lua/mediaplayer/services/youtube/init.lua

```diff
@@ -134,18 +82,127 @@ function SERVICE:GetMetadata( callback )
+
+				-- html couldn't be parsed
+				if not metadata.title or not isnumber(metadata.duration) then
+					callback(false, "Failed to parse HTML Page for metadata")
+					return
+				end
+
+				self:SetMetadata(metadata, true)
+
+				if self:IsTimed() then
+					MediaPlayer.Metadata:Save(self)
+				end
+			
+				callback(self._metadata)
 			end,
+			-- On failure
 			function( code )
 				callback(false, "Failed to load YouTube ["..tostring(code).."]")
-			end
+			end,
+			-- Headers
+			{
+				["User-Agent"] = "Googlebot"
+			}
 		)
+	end
+end
+
+---
+-- Get the value for an attribute from a html element
+-- 
+local function ParseElementAttribute( element, attribute )
+	if not element then return end
+	-- Find the desired attribute
+	local output = string.match( element, attribute.."%s-=%s-%b\"\"" )
+	if not output then return end
+	-- Remove the 'attribute=' part
+	output = string.gsub( output, attribute.."%s-=%s-", "" )
+	-- Trim the quotes around the value string
+	return string.sub( output, 2, -2 )
+end
+
+---
+-- Get the contents of a html element by removing tags
+-- Used as fallback for when title cannot be found
+--
+local function ParseElementContent( element )
+	if not element then return end
+	-- Trim start
+	local output = string.gsub( element, "^%s-<%w->%s-", "" )
+	-- Trim end
+	return string.gsub( output, "%s-</%w->%s-$", "" )
+end
+
+--  List of HTML entities to find in title and convert to their corresponding symbols
+local htmlEnts = {
+	["&quot;"] = "\"",
+	["&lt;"] = "<",
+	["&gt;"] = ">",
+	["&amp;"] = "&"
+}
 
+---
+-- Turn HTML entities into symbols
+--
+local function ParseTitleSymbols( string )
+	if not string then return end
+	local output = string
+	for entity, symbol in pairs( htmlEnts ) do
+		output = string.gsub( output, entity, symbol )
+	end
+	return output
+end
+
+-- Lua search patterns to find metadata from the html
+local patterns = {
+	["title"] = "<meta%sproperty=\"og:title\"%s-content=%b\"\">",
+	["title_fallback"] = "<title>.-</title>",
+	["thumb"] = "<meta%sproperty=\"og:image\"%s-content=%b\"\">",
+	["thumb_fallback"] = "<link%sitemprop=\"thumbnailUrl\"%s-href=%b\"\">",
+	["duration"] = "<meta%sitemprop%s-=%s-\"duration\"%s-content%s-=%s-%b\"\">",
+	["live"] = "<meta%sitemprop%s-=%s-\"isLiveBroadcast\"%s-content%s-=%s-%b\"\">",
+	["live_enddate"] = "<meta%sitemprop%s-=%s-\"endDate\"%s-content%s-=%s-%b\"\">"
+}
+
+---
+-- Function to parse video metadata straight from the html instead of using the API
+--
+function SERVICE:ParseYTMetaDataFromHTML( html, videoId )	
+	--MetaData table to return when we're done
+	local metadata = {}
+
+	-- Fetch title and thumbnail, with fallbacks if needed
+	metadata.title = ParseElementAttribute(string.match(html, patterns["title"]), "content") 
+		or ParseElementContent(string.match(html, patterns["title_fallback"]))
+
+	-- Parse HTML entities in the title into symbols
+	metadata.title = ParseTitleSymbols(metadata.title)
```

Comment:
I don't think the function is used elsewhere, but the project has a URL library to handle [decoding html entities.](https://github.com/samuelmaddock/gm-mediaplayer/blob/eca816079b67cb18defc475e67a6a9a173bf0ff2/lua/autorun/includes/extensions/sh_url.lua#L140)
```suggestion
	metadata.title = url.htmlentities_decode(metadata.title)
```